Add onAudioFocusGained to methodCallHandler on iOS.

snaeji opened the pull request:
I saw that the audioInterrupt AVAudioSessionInterruptionTypeEnded was already implemented in the DarwinAudioSession.m but was missing the corresponding methodHandler.

The implementation is just a copy paste of the onInterruptionEvent handler and works as intended.

ryanheise commented:
Oops, thanks for spotting this. I never intended for there to be two separate method keys for "onInterruptionEvent" AND "onAudioFocusGained". The latter was copied was a copy and paste job and I forgot to rename it. So the correct solution would be to edit `ios/Classes/DarwinAudioSession.m` to and rename the latter method key to the former, so they should all say "onInterruptionEvent". Then you would not need to copy&paste two identical switch cases on the Dart side.

snaeji commented:
That makes much more sense when reading the code again!

The least i could do, I was looking over this package and was really surprised by how advanced and next level it is.

I reverted and added a new commit with the right implementation.

ryanheise commented:
Merged, thanks!

> I was looking over this package and was really surprised by how advanced and next level it is.

There's a lot of commented out code which I eventually would like to support, such as the ability to query info about the attached audio device and other things. However, the immediate goal was to just get enough of the API implemented to solve the problem of activating and managing the audio session across multiple audio plugins.

ryanheise commented:
P.S. I assume so, but did you test this version of the PR and was it working as expected for you?

snaeji commented:
Cool! The commented code makes it easier for someone with very little obj-c knowledge to maybe pick some low hanging fruits.

Yeah sorry i should have mentioned it. Tested on iOS 13.7, iPhone SE (Second generation) and works as intended 🎉

ryanheise commented:
Thanks for confirming! I've just published 0.0.6 including your fix.
